### PR TITLE
Fix minor issues

### PR DESCRIPTION
**Pull request by Jentsch**
Hi! I found your project mentioned on the zio resources page ([1]). I had some minor problems in getting everything up and running. This changes should help the next person the get faster up to speed.

Thanks for providing this example project.

[1]: https://zio.dev/docs/resources/resources

### CODE REVIEW

File: server/src/main/scala/core/Core.scala [hunk 1]
```diff
@@ -23,7 +23,7 @@ import akka.actor.ActorSystem
  */
 trait Core {
 
-  implicit def actorSystem: ActorSystem
```
Comment (rleibman) [inline server/src/main/scala/core/Core.scala:26]: It's good practice to make your traits declare defs... that way your implementing classes can be vars or defs, but the opposite is not true.

File: server/src/main/scala/dao/LiveRepository.scala [hunk 2]
```diff
@@ -37,12 +37,11 @@ trait LiveRepository
     with DatabaseProvider
     with Config
     with ModelSlickInterop {
-  self =>
   private val runtime                               = new DefaultRuntime {}
   implicit val dbExecutionContext: ExecutionContext = runtime.platform.executor.asEC
 
   implicit def provideDB[R](dbio: DBIO[R]): IO[RepositoryException, R] =
-    fromDBIO(dbio).provide(self)
+    fromDBIO(dbio).provide(this)
```
Comment (rleibman) [inline server/src/main/scala/dao/LiveRepository.scala:44]: Hey, if this works... there must have been a reason I had self.

**Review by rleibman — CHANGES_REQUESTED** [review]

**Comment by rleibman** [comment]
BTW... thanks a lot for the comments and PR!

**Comment by Jentsch** [comment]
Oh, didn't saw that you already merge this. I'll open an other PR to respond to your comment about `def`s in traits.

**Pull request merged**